questions/javascript-questions.md

```diff
@@ -1,21 +1,19 @@
 # JS Questions:
 
-* Explain event delegation
-* Explain how `this` works in JavaScript
-* Explain how prototypal inheritance works
-* What do you think of AMD vs CommonJS?
-* Explain why the following doesn't work as an IIFE: `function foo(){ }();`.
-  * What needs to be changed to properly make it an IIFE?
+* Explain event delegation.
+* Explain how `this` works in JavaScript.
+  * Can you give an example of how `this` has changed in ES6?
+* Explain how prototypal inheritance works.
 * What's the difference between a variable that is: `null`, `undefined` or undeclared?
   * How would you go about checking for any of these states?
 * What is a closure, and how/why would you use one?
 * Can you describe the main difference between a `forEach` loop and a `.map()` loop and why you would pick one versus the other?
 * What's a typical use case for anonymous functions?
-* How do you organize your code? (module pattern, classical inheritance?)
```

Comment:
This could be brought back, without the "module pattern, classical inheritance" part because even in robust frameworks and libraries people still have to make choices about organization.

Reply:
disagree about module pattern - it both about loading modules and abstractions for me. Probably we need rephrase it